rad/src/lib.rs

```diff
@@ -306,7 +307,20 @@ async fn http_response(
             message: x.to_string(),
         })?;
 
-    let result = run_retrieval_with_data_report(retrieve, &response_string, context, settings);
+    let result = run_retrieval_with_data_report(retrieve, &response_string, context, settings).map(
+        |report| {
+            let completion_ts = std::time::SystemTime::now();
+            RadonReport {
```

Comment:
We normally put an empty line before values that are implicitly returned at the end of a function or closure.